src/domexpression.js

```diff
@@ -120,7 +120,17 @@ var _ = Mavo.DOMExpression = $.Class({
 			this.item.expressions.delete(this);
 		}
 
-		this.mavo.expressions.register(this);
+		let expressions = [...this.item.expressions].filter(e => e.attribute === this.attribute && e.active !== false);
+		if (expressions.some(e => e.originalAttribute?.startsWith("mv-attr-"))) {
+			for (let expression of expressions.filter(e => !e.originalAttribute?.startsWith("mv-attr-"))) {
+				// We already have expressions associated with this attribute; discard them since mv-attr-* has priority
+				expression.active = false;
+			}
+		}
+
+		if (this.active !== false) {
+			this.mavo.expressions.register(this);
+		}
```

Comment:
@LeaVerou, @adamjanicki2, I came up with this way of being deterministic when both `foo` and `mv-attr-foo` have an expression, and we expect `mv-attr-foo` to take precedence. Could you please review it?

Reply:
What happened before? Did it follow the order of attributes or was it non-deterministic?

Reply:
As I learned from my experiments, it did follow the order of attributes. However, I can’t guarantee that it was deterministic all the time. I believe some inconsistency might appear because of race conditions. But I’m not sure. That’s why I’m leaning towards the `mv-attr-*` wins route.

Reply:
I think a better route would be to prevent the creation of these useless objects altogether, rather than having to make them inactive later. I think a better place for this logic might be in `extract()` or `traverse()` in [expressions.js](https://github.com/mavoweb/mavo/blob/master/src/expressions.js), especially since that's where other conditions for ignoring attributes are handled (e.g. `mv-expressions-ignore`).

Reply:
Good idea. I struggled to find an appropriate place for the logic I meant to implement—I tried to prevent the creation of the expressions we didn't mean to use but without luck. I will commit another try, but in `expressions.js` this time. Thank you!